### ISSUE TITLE
authenticate: Field 'user' of type 'User' must have a sub selection

### ISSUE THREAD

**Issue by artemzakharov** [LawJolla/prisma-auth0-example#6]
Thanks for providing this great example! I'm trying to use it in my own server, but I get the following error when I run the 'authenticate' mutation:

```
query ($_where: UserWhereUniqueInput!) {
  user(where: $_where)
}
operationName: null
variables:
{
  "_where": {
    "auth0id": "1580459988643362"
  }
}
[GraphQL error]: Message: Field 'user' of type 'User' must have a sub selection. (line 2, column 3):
  user(where: $_where)
  ^, Location: [object Object], Path: undefined
[Network error]: Error: Field 'user' of type 'User' must have a sub selection. (line 2, column 3):
  user(where: $_where)
```

This is the line that errors out:

`let user = await ctx.db.query.user({ where: { auth0id } }, info);`

And this is my mutation: 

```
mutation ($idToken: String!) {
  authenticate(idToken: $idToken) {
    user {
      id
    }
  }
}
```
Any idea what's going on here? I'm still fresh from the GCF and this is all incredibly confusing. I did notice that taking out the `info` param made the query work, but then you can't control what gets returned, right?

**Comment by LawJolla** [comment]
Hey @artemzakharov ,

Thanks for stopping by and the question!

`authenticate` returns a `User` node, so you can't call for the User type in the return.  It's like you're calling it twice.

Try:

```
mutation ($idToken: String!) {
  authenticate(idToken: $idToken) {
      id
  }
}
```

**Comment by artemzakharov** [comment]
@LawJolla Thank you, I just realized what the problem was! I was trying to integrate your code into the Typescript starter, and I didn't notice that I arranged my schema such that the `authenticate` mutation was returning the `AuthPayload` type that came with the boilerplate instead of just a `User`! `AuthPayload` has a `user` and another parameter so my playground showed no errors, hence all the confusion. Wish errors like this were easier to diagnose.
